frontend/src/hooks/useNavigation.tsx

```
@@ -97,34 +139,77 @@ export function NavigationProvider({ children }: { children: ReactNode }) {
+      // visit before it corrects. Only when the URL named something else, so a
+      // bare `/` stays bare.
+      if (urlWorldViewId !== null && urlWorldViewId !== worldView.id) {
+        syncWorldViewParam(worldView);
       }
     }
-  // eslint-disable-next-line react-hooks/exhaustive-deps -- only depend on length to avoid re-selecting on every fetch
-  }, [worldViews.length, isAdmin]);
+  // Keyed on the ids rather than the array: the query returns a new array on
+  // every fetch, and length alone would miss a same-size list with different
+  // members.
+  // eslint-disable-next-line react-hooks/exhaustive-deps -- worldViews by id, not identity
```

ESLint

[error] Definition for rule 'react-hooks/exhaustive-deps' was not found.

(react-hooks/exhaustive-deps)